.github/workflows/ci.yml

```diff
@@ -56,7 +56,8 @@ jobs:
             ~/.cargo/git/db
             # contains package information of crates installed via `cargo install`.
             ~/.cargo/.crates.toml
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+            ~/.cargo/.crates2.json
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}-xargo0.3.25
```

Comment:
We could compute the cache name by fetching the version id of the latest xargo. This way we'd get auto updates.

Alternatively we can assert in ci that we're using the same xargo version as the cache suggests (and that it is the latest). That way CI would fail when the xargo version is outdated)

Reply:
Or we just actually put somewhere (into a variable) the version of xargo we want to install. :shrug:  But then we have to remember to bump that when xargo changes. In terms of reproducibility that is nice, but for mere utilities it is a bit annoying...

But we can do that another time IMO. I could back out the CI changes entirely from this PR if you prefer.

Reply:
Nah, let's do this in follow ups. Once I get the ui stuff done i'll try to spend more time on our CI